main.js

```diff
@@ -0,0 +1,280 @@
+// const axios = require('axios')
+
+let countries = [];
+let APIerror = "";
+
+// delete a DOMtree, including the parent element in the parameter to this function
+
+function delDomTree(root) {
+    for (let i = 0; i < root.children.length; i++) {
+        delDomTree(root.children[i]);
+    }
+    root.remove();
+}
+
+// if string is undefined, empty or contains less than 4 characters
+// return the empty message
+
+function cleanString(string, emptyMessage) {
+
+    let out = string ?? emptyMessage;
+    if (out.includes('[') ) out = emptyMessage;
+    return out;
+
+}
+
+
+// format an array of objects with the name attribute to a comma
+// separated string with last 2 element separated by 'and' in stead of ','.
+// so results:
+// one -> one
+// one,two -> one and two
+// one,two,thrre -> one,two and three.
+
+function A2String(A) {
```

Comment:
Variablen én functienamen schrijf je onderandere in javascript camelCase. Geef ze daarnaast ook een goeie en duidelijke naam zodat je kan herkennen wat er gebeurt